src/lib/storage/abstract_device_factory.rb

```diff
@@ -163,8 +165,12 @@ def build_tree_recursive(parent, name, content)
         end
       end
 
+      # rubocop:disable Lint/UselessAccessModifier
+
       protected
 
+      # rubocop:enable Lint/UselessAccessModifier
+
```

Comment:
actually what is a goal of this protected? Also I think that documentation is at wrong place. Where yardoc will render it? I think for such stuff documentation have to be in class documentation itself so when you subclass you usually read top level documentation. Another way how to do it, is to make real methods and use e.g. abstract_method gem that is already used by yast2 to mark methods as purely abstract.